source/Module/Rule.ProcessMitigation/Convert/Methods.ps1

```diff
@@ -150,60 +116,143 @@ function Test-PoliciesToEnable
-function Split-ProcessMitigationRule
+function Get-MitigationValue
 {
     [CmdletBinding()]
-    [OutputType([array])]
+    [OutputType([string])]
     param
     (
         [Parameter(Mandatory = $true)]
         [AllowEmptyString()]
-        [string]
-        $MitigationTarget
+        [string[]]
+        $CheckContent
     )
 
-    return ( $MitigationTarget -split ',' )
+    Write-Verbose "[$($MyInvocation.MyCommand.Name)]"
+
+    try
+    {
+        $mitigationValue = ($CheckContent | Select-String -Pattern $regularExpression.MitigationValue -CaseSensitive).Matches.Value
+
+        if ($mitigationValue -match 'ON|True')
+        {
+            $mitigationValue = 'true'
+        }
+        else
+        {
+            $mitigationValue = 'false'
+        }
+
+        return $mitigationValue
+    }
+    catch
+    {
+        Write-Verbose "[$($MyInvocation.MyCommand.Name)] Mitigation Value : Not Found"
+        return $null
+    }
 }
 
 <#
     .SYNOPSIS
         Check if the string (MitigationTarget) contains a comma. If so the rule needs to be split
 #>
-function Test-MultipleProcessMitigationRule
+
+function Test-MultipleProcessMitigations
 {
     [CmdletBinding()]
     [OutputType([bool])]
     param
     (
         [Parameter(Mandatory = $true)]
-        [AllowEmptyString()]
-        [string]
-        $MitigationTarget
+        [psobject]
+        $CheckContent
     )
 
-    if ( $MitigationTarget -match ',')
+    $matchTargets = ($CheckContent | Select-String -Pattern $regularExpression.MitigationTarget -AllMatches).Matches.Value | Select-Object -Unique
+    $matchTypes = ($CheckContent | Select-String -Pattern $regularExpression.MitigationType -AllMatches).Matches.Value | Select-Object -Unique
+    $matchNames = ($CheckContent | Select-String -Pattern $regularExpression.MitigationName -AllMatches).Matches.Value | Select-Object -Unique
+
+    if (($matchTargets.count -gt 1) -or ($matchTypes.count -gt 1) -or ($matchNames.count -gt 1))
     {
         return $true
     }
     return $false
 }
+
+
+
+<#
+    .SYNOPSIS
+        Consumes a list of mitigation targets seperated by a comma and outputs an array
+#>
+function Split-MultipleProcessMitigations
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.ArrayList])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [psobject]
+        $CheckContent
+    )
+
+    $matchNamesGroup = @()
+    $processMitigations = @()
+    $matchTargets = ($CheckContent | Select-String -Pattern $regularExpression.MitigationTarget -AllMatches).Matches.Value | Select-Object -Unique
+
+    if($matchTargets -eq "[application name]")
+    {
+        $matchTargets = ((($CheckContent | Select-String -Pattern ".*.EXE|.*.exe" -CaseSensitive).Matches.Value) -split (",")).replace("and ", "")
+    }
+
+    $matchTypes = ($CheckContent | Select-String -Pattern $regularExpression.MitigationType -AllMatches).Matches.Value | Select-Object -Unique
+
+    foreach($mitigationTarget in $matchTargets)
```

Comment:
Should be a space after the foreach keyword.

Reply:
Done